src/plugin.cpp

```diff
@@ -1,36 +1,94 @@
 #include <orthanc/OrthancCPlugin.h>
+#include <string>
+#include <regex>
 #include "configuration.h"
 
-static OrthancPluginContext* context_ = nullptr;
+static OrthancPluginContext *context_ = nullptr;
 
-extern "C" {
-    int32_t OrthancPluginInitialize(OrthancPluginContext* context){
-        context_ = context;
-
-        /* Check the version of the Orthanc core */
-        if (OrthancPluginCheckVersion(context) == 0){
-            char info[256];
-            sprintf(info, "Your version of Orthanc (%s) must be above %d.%d.%d to run this plugin",
-                    context->orthancVersion,
-                    ORTHANC_PLUGINS_MINIMAL_MAJOR_NUMBER,
-                    ORTHANC_PLUGINS_MINIMAL_MINOR_NUMBER,
-                    ORTHANC_PLUGINS_MINIMAL_REVISION_NUMBER);
-
-            OrthancPluginLogError(context, info);
-            return -1;
-        }
-        return 0;
-    }
-    
-    void OrthancPluginFinalize(){
+OrthancPluginErrorCode OnStoredCallback(OrthancPluginDicomInstance *instance, const char *instanceId) {
+    char buffer[1024];
+    sprintf(buffer, "Just received a DICOM instance of size %d and ID %s from origin %d (AET %s)",
+            (int) OrthancPluginGetInstanceSize(context_, instance), instanceId,
+            OrthancPluginGetInstanceOrigin(context_, instance),
+            OrthancPluginGetInstanceRemoteAet(context_, instance));
+    OrthancPluginLogWarning(context_, buffer);
 
-    }
+    std::string instanceId_str = std::string(instanceId);
+
+    OrthancPluginErrorCode error;
+
+    OrthancPluginMemoryBuffer anonymized_buffer;
+    std::string anonymize_uri = "/instances/" + instanceId_str + "/anonymize";
+    std::string anonymize_body = "{}";
+    error = OrthancPluginRestApiPost(context_, &anonymized_buffer, anonymize_uri.c_str(), anonymize_body.c_str(),
+                                     anonymize_body.length());
+    if (error) return error;
 
-    const char* OrthancPluginGetName(){
-        return ORTHANC_PLUGIN_NAME;
+    OrthancPluginMemoryBuffer instance_buffer;
+    std::string instance_uri = "/instances/" + instanceId_str;
+    error = OrthancPluginRestApiGet(context_, &instance_buffer, instance_uri.c_str());
+    if (error) return error;
+
+    std::string instance_json = std::string((char *) instance_buffer.data);
+    std::regex uuid_regex("([a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12})");
+    std::smatch uuid_match;
+    std::string file_uuid;
+    if (std::regex_search(instance_json, uuid_match, uuid_regex)) {
```

Comment:
If this data really is json, it should be cheaper to parse the json and access the uuid directly. I would expect anyways.

Reply:
That's the plan, this is a band-aid solution until we get the libraries working in a plugin environment